### PR TITLE
Adding CI Pipeline

### PR DESCRIPTION
**Pull request by ccmagruder**

### CODE REVIEW

File: .runner/Dockerfile [hunk 1]
```diff
@@ -1,9 +1,13 @@
-FROM ubuntu:24.04
+FROM ccmagruder/devcontainer-gemm:latest
+# FROM nvidia/cuda:12.6.2-cudnn-devel-ubuntu24.04
+# FROM ubuntu:24.04
 
 ARG RUNNER_VERSION="2.321.0"
 
 ARG DEBIAN_FRONTEND=noninteractive
 
+COPY start.sh start.sh
```
Comment (jeffrey-hokanson) [inline .runner/Dockerfile:9]: Do you want an absolute path in the container side?

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+RUN apt update -y && apt upgrade -y \
+  && apt install -y --no-install-recommends \
+    build-essential \
+    clang \
+    cmake \
+    gcc \
+    git \
+    libbenchmark-dev \
+    libgtest-dev
```
Comment (jeffrey-hokanson) [inline .devcontainer/Dockerfile]: I think we should clear the apt-cache to minimize layer size.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -7,12 +7,26 @@ on:
...
+        run: cmake -DCMAKE_BUILD_TYPE=Release -S gemm -B build .
+
+      - name: cmake build
+        run: cmake --build build
+
+      - name: google test
+        run: ctest --test-dir build --output-on-failure
+
+      - name: google benchmark
+        run: build/tests/bMatrix
```
Comment (jeffrey-hokanson) [inline .github/workflows/main.yml:32]: It's been a while since I've looked at GitHub workflows. Is this running all in one step?

**Review by jeffrey-hokanson — COMMENTED** [review]
Pretty much everything looks good. I'm looking forward to talking through the architecture.

File: .devcontainer/Dockerfile (hunk 2 above)
Comment (ccmagruder) [reply .devcontainer/Dockerfile]: Resolved in #4, thanks for the changes!

**Pull request merged**